tests/dispatch/utils.py

```diff
@@ -118,5 +128,21 @@ def stream_complete(self, stream_id):
         response_headers = [(b":status", status_code_bytes)] + response.headers.raw
 
         self.conn.send_headers(stream_id, response_headers)
-        self.conn.send_data(stream_id, response.content, end_stream=True)
-        self.buffer += self.conn.data_to_send()
+        asyncio.ensure_future(self.send_data(response.content, stream_id))
+
+    async def send_data(self, data, stream_id):
+        window_size = self.conn.local_flow_control_window(stream_id)
+        max_frame_size = self.conn.max_outbound_frame_size
+        chunk_size = min(len(data), window_size, max_frame_size)
+
+        for idx in range(0, len(data), chunk_size):
+            left_window_size = self.conn.local_flow_control_window(stream_id)
+            if left_window_size < chunk_size:
+                self.wait_events[stream_id] = asyncio.Event()
```

Comment:
This may have to be dealt with within #217 once it's merged, just cc @florimondmanca

Reply:
@sethmlarson Yup; I haven't looked at genericizing the HTTP/2 mocks in #217 yet, so thanks for the notice.